### ISSUE TITLE
Extend browser support.

### ISSUE THREAD

**Issue by dbuenzli** [dbuenzli/webbrowser#1]
For now the full browser reload strategy is only supported on osx with chrome and safari. Help for extending support along [the documentation](http://erratique.ch/software/webbrowser/doc/Webbrowser.html) is welcome. 
- macosx support for firefox and other browsers
- linux support beyond `xdg-open`  
- windows everything.

**Comment by apatil** [comment]
@dbuenzli a slightly different approach in https://github.com/apatil/ocaml-webview should be cross-platform, at least in theory. Would that address the use case for which you wrote this library, or do you need a full-featured browser with forward and back buttons etc.?

**Comment by dbuenzli** [comment]
No. I want that to open/reload in the user's browser.

**Comment by apatil** [comment]
Got it.

**Comment by Chris00** [comment]
I think [this approach](https://blog.markvincze.com/programmatically-refreshing-a-browser-tab-from-a-golang-application/) should be used.

**Comment by dbuenzli** [comment]
> I think [this approach](https://blog.markvincze.com/programmatically-refreshing-a-browser-tab-from-a-golang-application/) should be used.

That doesn't allow you to reload arbitrary pages. You need to have the script on your page.

**Comment by Chris00** [comment]
Right.  You have to send a local page with the script  `<iframe src="`_URI_`" mozbrowser remote>`.

**Comment by Chris00** [comment]
I suppose something similar is possible for other browsers.

**Comment by Chris00** [comment]
Generating the page wrapper also allows us to programmatically choose the port (since it must be in the web page) and also have a single websocket server for several URL (each page connecting to a different path on websocket server) that could automatically die if there are no connections for a given time from both applications and web pages.

**Comment by Chris00** [comment]
I forgot to point to the [Firefox documentation](https://developer.mozilla.org/en-US/docs/Web/API/Using_the_Browser_API).

**Comment by Chris00** [comment]
I suppose JS could also reload the whole page using [`document.location.reload(true);`](https://developer.mozilla.org/en-US/docs/Web/API/Location/reload) — that should work on Chrome and Edge too.

**Comment by dbuenzli** [comment]
I'm not sure exactly how it can achieve what it's doing now (which is to reload any tab you might have in your browser) also TBH that feels like killing a fly with an atomic bomb.

**Comment by Chris00** [comment]
> how it can achieve what it's doing now (which is to reload any tab

It can't reload any tab (because that requires browser scripting which does not seem to be readily available — short of writing a browser plugin) but it can [reload tabs that were opened through the library](https://blog.markvincze.com/programmatically-refreshing-a-browser-tab-from-a-golang-application/), which is usually what one wants (it's not bad not to touch existing tabs).

>  that feels like killing a fly with an atomic bomb.

I agree and there may be other solutions but, in the absence of browser scripting, I guess they will involve JS talking to a small local server — maybe one can use a simpler one than a websocket.

**Comment by Chris00** [comment]
Note that [livereload](http://livereload.com/) uses the above technology to reload tabs when the underlying file is updated, so I'm not very confident there is a _much simpler_ solution...

**Comment by dbuenzli** [comment]
This package is only maintained, no longer developed. Similar functionality can be found in the [b0 project](https://erratique.ch/software/b0).
